src/flp/types/fixedpoint_l2.rs

```diff
@@ -323,21 +346,70 @@ where
             gadget1_chunk_len,
         })
     }
+
+    /// This noising function can be called on the aggregation result to make
+    /// the entire aggregation process differentially private. The noise is
+    /// calibrated to result in a guarantee of `1/2 * epsilon^2` zero-concentrated
+    /// differential privacy, where `epsilon` is given by `dp_strategy.budget`.
+    fn add_noise<R: Rng>(
```

Comment:
How about just moving this code to `add_noise_to_result()`?

Reply:
That's not quite possible: For testing purposes we need a deterministic function taking an `Rng` as argument. The `TypeWithNoise::add_noise_to_result()` method does not have such an argument. In the implementation of `add_noise_to_result()` we create an `Rng` from system noise, and pass it to `add_noise()`.

This setup was advised for by @divergentdave [here](https://github.com/divviup/libprio-rs/pull/578#issuecomment-1622009942).